Add new sample: no-cookies rule manager (#986)
GoogleChrome/chrome-extensions-samples: functional-samples/sample.dnr-rule-manager/README.md

# chrome.declarativeNetRequest - No Cookies Rule Manager

> This sample demonstrates using the [`chrome.declarativeNetRequest`](https://developer.chrome.com/docs/extensions/reference/declarativeNetRequest/) API to remove the "Cookie" header from requests matching custom url filters.

Reviewer: ```suggestion
This sample demonstrates using the [`chrome.declarativeNetRequest`](https://developer.chrome.com/docs/extensions/reference/declarativeNetRequest/) API to remove the `Cookie` header from requests matching custom URL filters.
```

## Overview

> Once this extension is installed, there will be two default rules:

Reviewer: ```suggestion
This extension has two default rules:
```

> 1. Any main frame or XHR requests ending with `?nocookies=1` will have their cookies removed (URL Filter).
> 
> 2. Any main frame or XHR requests matching `.*\.google\.com` will have their cookies removed (Regex Filter).

Reviewer: ```suggestion
* Any main frame or XHR requests ending with `?nocookies=1` will have their cookies removed (URL Filter).

* Any main frame or XHR requests matching `.*\.google\.com` will have their cookies removed (Regex Filter).
```

Reviewer: Corporate styleguide reserves numbers for procedures.

> For example, install this extension and try navigating to <https://github.com/GoogleChrome/chrome-extensions-samples?no-cookies=1> or <https://www.google.com> You should appear signed out. The number of requests modified by this extension will be displayed on the extension's badge.

Reviewer 2: ```suggestion
For example, install this extension and try navigating to <https://github.com/GoogleChrome/chrome-extensions-samples?no-cookies=1> or <https://www.google.com> - you should appear signed out. The number of requests modified by this extension will be displayed on the extension's badge.
```

You can edit these rules in the manager page.

## Implementation Notes

> [`chrome.declarativeNetRequest.setExtensionActionOptions`](https://developer.chrome.com/docs/extensions/reference/declarativeNetRequest/#method-setExtensionActionOptions) is used to set the extension's badge text to the number of requests modified by the extension.

Reviewer: ```suggestion
[`chrome.declarativeNetRequest.setExtensionActionOptions()`](https://developer.chrome.com/docs/extensions/reference/declarativeNetRequest/#method-setExtensionActionOptions) sets the extension's badge text to the number of requests modified by the extension.
```

> `declarativeNetRequestFeedback` permission is required to show the matching history.

Reviewer: ```suggestion
`"declarativeNetRequestFeedback"` permission is required to show the matching history.
```

> ## Running this extension

Reviewer: ```suggestion
## Run this extension
```

Reviewer: I can't remember how other samples phrase this. My suggestion is required by the corporate style guide. Ignore it if it's not consistent with other READMEs.
